api/src/main/java/javax/enterprise/inject/spi/builder/AnnotatedTypeConfigurator.java

```diff
@@ -0,0 +1,422 @@
+                                                           int position,
+                                                           Annotation annotation);
+
+    /**
+     * Remove an annotation from the specified constructor parameter.
+     *
+     * @param constructor    the constructor to remove the annotation from
+     * @param position       the position of the parameter to remove
+     * @param annotationType the annotation type to remove
+     * @throws IllegalArgumentException if the annotationType is null, if the
+     *                                  constructor is not currently declared on the type or if the
+     *                                  parameter is not declared on the constructor
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromConstructorParameter(Constructor<T> constructor,
+                                                                int position,
+                                                                Class<? extends Annotation> annotationType);
+
+    /**
+     * Remove an annotation from the specified parameter.
+     *
+     * @param parameter      the parameter to remove the annotation from
+     * @param annotationType the annotation type to remove
+     * @throws IllegalArgumentException if the annotationType is null, if the
+     *                                  callable which declares the parameter is not currently declared
+     *                                  on the type or if the parameter is not declared on either a
+     *                                  constructor or a method
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromParameter(AnnotatedParameter<? super T> parameter,
+                                                     Class<? extends Annotation> annotationType);
+
+    /**
+     * Add an annotation to the specified parameter. If the callable which
+     * declares the parameter is not already present, it will be added. If the
+     * parameter is not already present on the callable, it will be added.
+     *
+     * @param parameter  the parameter to add the annotation to
+     * @param annotation the annotation to add
+     * @throws IllegalArgumentException if the annotation is null or if the
+     *                                  parameter is not declared on either a constructor or a method
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> addToParameter(AnnotatedParameter<? super T> parameter, Annotation annotation);
+
+    /**
+     * Remove annotations from the type, and all of it's members. If an
+     * annotation of the specified type appears on the type declaration, or any
+     * of it's members it will be removed.
+     *
+     * @param annotationType the type of annotation to remove
+     * @throws IllegalArgumentException if the annotationType is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromAll(Class<? extends Annotation> annotationType);
```

Comment:
Variant of this method which accepts `Annotation` parameter could be helpful here as well. It's questionable in case of other remove* methods - it's probably not needed....